File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */

class Solution {
    fun maxDepth(root: TreeNode?, depth: Int = 0): Int {
        if (root == null) return depth
        
        return Math.max(
            maxDepth(root.left, depth + 1), 
            maxDepth(root.right, depth + 1)
        )
    }
}

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        # 왼쪽 자식과 오른쪽 자식에서 높이를 구한 후, 둘의 max값 리턴
        def helper(root, depth):
            if not root:
                return depth

            left = helper(root.left, depth + 1)
            right = helper(root.right, depth + 1)

            return max(left, right)

        return helper(root, 0)
